### ISSUE TITLE
Openssl 0.7.14 fails to build

### ISSUE THREAD

**Issue by CorruptComputer** [zeyla/serenity#56]
Going [here](https://github.com/sfackler/rust-openssl) and compiling 0.9.5 succeeds, however this older version doesn't seem to work. Any way to update the dependencies so that it has this newer version?

Error:
````
Build failed, waiting for other jobs to finish...
error: failed to run custom build command for `openssl v0.7.14`
process didn't exit successfully: `C:\Users\ngupton\Desktop\serenity\target\debug\build\openssl-5464f8f6e728c35a\build-script-build` (exit code: 101)
--- stdout
TARGET = Some("x86_64-pc-windows-msvc")
OPT_LEVEL = Some("0")
PROFILE = Some("debug")
TARGET = Some("x86_64-pc-windows-msvc")
debug=true opt-level=0
HOST = Some("x86_64-pc-windows-msvc")
TARGET = Some("x86_64-pc-windows-msvc")
TARGET = Some("x86_64-pc-windows-msvc")
HOST = Some("x86_64-pc-windows-msvc")
CC_x86_64-pc-windows-msvc = None
CC_x86_64_pc_windows_msvc = None
HOST_CC = None
CC = None
TARGET = Some("x86_64-pc-windows-msvc")
HOST = Some("x86_64-pc-windows-msvc")
CFLAGS_x86_64-pc-windows-msvc = None
CFLAGS_x86_64_pc_windows_msvc = None
HOST_CFLAGS = None
CFLAGS = None
running: "C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\bin\\amd64\\cl.exe" "/nologo" "/MD" "/Z7" "/I" "C:\\OpenSSL-Win64\\include\\" "/FoC:\\Users\\ngupton\\Desktop\\serenity\\target\\debug\\build\\openssl-5464f8f6e728c35a\\out\\src\\c_helpers.o" "/c" "src/c_helpers.c"
c_helpers.c
C:\OpenSSL-Win64\include\openssl/lhash.h(198): warning C4090: 'function': different 'const' qualifiers
src/c_helpers.c(4): error C2037: left of 'references' specifies undefined struct/union 'ssl_st'
src/c_helpers.c(4): error C2065: 'CRYPTO_LOCK_SSL': undeclared identifier
src/c_helpers.c(8): error C2037: left of 'references' specifies undefined struct/union 'ssl_ctx_st'
src/c_helpers.c(8): error C2065: 'CRYPTO_LOCK_SSL_CTX': undeclared identifier
src/c_helpers.c(12): error C2037: left of 'references' specifies undefined struct/union 'x509_st'
src/c_helpers.c(12): error C2065: 'CRYPTO_LOCK_X509': undeclared identifier
ExitStatus(ExitStatus(2))


command did not execute successfully, got: exit code: 2



--- stderr
thread 'main' panicked at 'explicit panic', C:\Users\ngupton\.cargo\registry\src\github.com-1ecc6299db9ec823\gcc-0.3.41\src\lib.rs:1018
note: Run with `RUST_BACKTRACE=1` for a backtrace.
````

**Comment by fwrs** [comment]
i don't think that's an issue with serenity

**Comment by GetRektByMe** [comment]
It's not an issue w/ the lib

**Comment by zeyla** [comment]
I'll look into updating the required rust-openssl version in the next couple days. The short summary is switching from hyper to reqwest, which makes use of rust-native-tls rather than rust-openssl. Monday, 16 January 2017, 09:57AM -08:00 from CorruptComputer  notifications@github.com :

>Going  here and compiling 0.9.5 succeeds, however this older version doesn't seem to work. Any way to update the dependencies so that it has this newer version?
>Error:
>Build failed, waiting for other jobs to finish...
error: failed to run custom build command for `openssl v0.7.14`
process didn't exit successfully: `C:\Users\ngupton\Desktop\serenity\target\debug\build\openssl-5464f8f6e728c35a\build-script-build` (exit code: 101)
--- stdout
TARGET = Some("x86_64-pc-windows-msvc")
OPT_LEVEL = Some("0")
PROFILE = Some("debug")
TARGET = Some("x86_64-pc-windows-msvc")
debug=true opt-level=0
HOST = Some("x86_64-pc-windows-msvc")
TARGET = Some("x86_64-pc-windows-msvc")
TARGET = Some("x86_64-pc-windows-msvc")
HOST = Some("x86_64-pc-windows-msvc")
CC_x86_64-pc-windows-msvc = None
CC_x86_64_pc_windows_msvc = None
HOST_CC = None
CC = None
TARGET = Some("x86_64-pc-windows-msvc")
HOST = Some("x86_64-pc-windows-msvc")
CFLAGS_x86_64-pc-windows-msvc = None
CFLAGS_x86_64_pc_windows_msvc = None
HOST_CFLAGS = None
CFLAGS = None
running: "C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\bin\\amd64\\cl.exe" "/nologo" "/MD" "/Z7" "/I" "C:\\OpenSSL-Win64\\include\\" "/FoC:\\Users\\ngupton\\Desktop\\serenity\\target\\debug\\build\\openssl-5464f8f6e728c35a\\out\\src\\c_helpers.o" "/c" "src/c_helpers.c"
c_helpers.c
C:\OpenSSL-Win64\include\openssl/lhash.h(198): warning C4090: 'function': different 'const' qualifiers
src/c_helpers.c(4): error C2037: left of 'references' specifies undefined struct/union 'ssl_st'
src/c_helpers.c(4): error C2065: 'CRYPTO_LOCK_SSL': undeclared identifier
src/c_helpers.c(8): error C2037: left of 'references' specifies undefined struct/union 'ssl_ctx_st'
src/c_helpers.c(8): error C2065: 'CRYPTO_LOCK_SSL_CTX': undeclared identifier
src/c_helpers.c(12): error C2037: left of 'references' specifies undefined struct/union 'x509_st'
src/c_helpers.c(12): error C2065: 'CRYPTO_LOCK_X509': undeclared identifier
ExitStatus(ExitStatus(2))


command did not execute successfully, got: exit code: 2



--- stderr
thread 'main' panicked at 'explicit panic', C:\Users\ngupton\.cargo\registry\src\github.com-1ecc6299db9ec823\gcc-0.3.41\src\lib.rs:1018
note: Run with `RUST_BACKTRACE=1` for a backtrace.

**Comment by wsremand** [comment]
Just wanted to let you know that this still happens, I'm a total noob at rust so I can't quite seem to figure out how to bypass this issue. Log attached.
[output.txt](https://github.com/zeyla/serenity/files/1011935/output.txt)

**Comment by Flat** [comment]
You need to have Openssl-1.0 installed which will depend on your distro how you do that. You  then need to set `OPENSSL_LIB_DIR=path/to/lib/openssl-1.0` and `OPENSSL_INCLUDE_DIR=path/to/include/openssl-1.0` and then `cargo clean` and then you should be able to build as long as those envars are set.

**Comment by wsremand** [comment]
Thanks Flat!

I was looking in all the wrong places, but that worked. :)

**Comment by lolzballs** [comment]
@Flat I still have the same errors after doing that. I did some digging and 0.7.14 uses a `DEP_OPENSSL_INCLUDE` var, but it still doesn't work when I set that as well.

**Comment by bbigras** [comment]
It would be easier on Windows if every dependencies would use native-tls (most crate are switching to it) since it would use schannel instead of openssl.

**Comment by zeyla** [comment]
That's currently the plan with an upgrade to rust-websocket v0.20, I (or someone else) just need to find the time for it. I currently have a few other half- or mostly-finished updates that I want to push out first.

**Comment by zeyla** [comment]
Closed by 8f8a059. rust-websocket has been upgraded to v0.20 and now makes use of [`native-tls`](https://github.com/sfackler/rust-native-tls).
